### DISCUSSION TITLE
GitOps conformance and certification

### DISCUSSION THREAD

**Discussion by scottrigby** [open-gitops/project#105] [Ideas]
Our [description of the OpenGitOps project](https://github.com/open-gitops/project/blob/main/README.md#opengitops) mentions conformance and certification.

The working group has discussed in meetings some considerations around this, but we haven't documented any plans or next steps.

What might this look like?

**Comment by scottrigby** [comment]
A few thoughts on conformance.

The main consideration discussed so far has been that GitOps is a _journey_, not a one-size-fits-all endpoint. Conformance could align with the principles, and we have discussed that this could somehow be represented as a degree of compliance – per principle, and with some kind of qualitative description – rather than a binary Yes or No. Something like "to what degree is principle 1 followed? Explain in 1-2 sentences."

Perhaps also "are there required configurations or other considerations to keep in mind in order to use this tool or integration, in order to comply with each principle?" This relates to the best practices section that has yet to be added to the documents repo, but also could be asked specifically in connection with conformance.

Another idea that was mentioned previously is that an official self-assessment offered by OpenGitOps could be beneficial to end users. Something like a "GitOps maturity model," which feels somehow related here.

**Reply by cwstrommer** [reply]
> official self-assessment

I like this.  I myself have been toying with the idea of a set of questions to ask GitOps adopters, not so much to guide them **towards** recommended practices but more to guide them **away** from common fallacies.  Things such as "Have you considered how you're going to manage multiple version releases of the same application?"

**Comment by scottrigby** [comment]
Update on certification:
- https://github.com/open-gitops/project/discussions/156

**Comment by scottrigby** [comment]
PR in progress about conformance/compliance:
- https://github.com/open-gitops/project/pull/173
